### PR TITLE
Support invoices with prefix

### PR DESCRIPTION
**Pull request by bhelx**
Addresses issue https://github.com/killbilling/recurly-java-library/issues/147

Do we want to keep the old method and deprecate it? Should we have both methods supported and just add some more documentation into the methods?

### CODE REVIEW

File: src/test/java/com/ning/billing/recurly/TestRecurlyClient.java [hunk 1]
```diff
@@ -1210,7 +1210,9 @@ public void testCreateTrialExtensionCoupon() throws Exception {
             final Subscription subscription = recurlyClient.createSubscription(subscriptionData);
 
             Assert.assertNotNull(subscription);
-            Assert.assertEquals(subscription.getTrialEndsAt().getMonthOfYear(), now.getMonthOfYear() + 3);
+
+            final int expectedMonth = (now.getMonthOfYear() + 3) % 12;
+            Assert.assertEquals(subscription.getTrialEndsAt().getMonthOfYear(), expectedMonth);
```
Comment (bhelx) [inline src/test/java/com/ning/billing/recurly/TestRecurlyClient.java]: This just came up now that it's october (month 10). Oversight on my part :)

File: src/main/java/com/ning/billing/recurly/RecurlyClient.java [hunk 2]
```diff
@@ -577,6 +577,19 @@ public Invoice getInvoice(final Integer invoiceId) {
...
+     * Lookup an invoice (with or without prefix)
+     * <p>
+     * Returns the invoice, the invoice may or
+     * may not have a country code prefix (ex: IE1023)
+     *
+     * @param invoiceId String Recurly Invoice ID
+     * @return the invoice
+     */
+    public Invoice getInvoice(final String invoiceId) {
+        return doGET(Invoices.INVOICES_RESOURCE + "/" + invoiceId, Invoice.class);
```
Comment (pierre) [inline src/main/java/com/ning/billing/recurly/RecurlyClient.java]: Maybe `getInvoice(final Integer invoiceId)` could call that implementation?

**Comment by bhelx** [comment]
Okay, I'm going to keep both methods and try to make sure the documentation is clear

**Comment by bhelx** [comment]
Okay this has been updated

**Comment by pierre** [comment]
🎉

**Pull request merged**